# Use this notbook for your 3 choices of APIs

## API n° 1 : Inshorts informations (sciences)

In [61]:
import requests
import pandas as pd 
import numpy as np 
from pandas import json_normalize

In [62]:
res = requests.get('https://inshorts.deta.dev/news?category=science')
res

<Response [200]>

In [63]:
sc = res.json()
scdata = pd.DataFrame(sc)
# print(sc)
# scdata

In [64]:
flattened_data = json_normalize(scdata['data'])
# flattened_data


In [65]:
flattened_data.shape

(25, 9)

## My interesting output from this API

In [69]:
print(flattened_data['title'][16])
print(flattened_data['content'][16])


James Webb's images show 'barred' galaxies from 8-11 bn years ago
Images from NASA's James Webb Telescope have revealed galaxies with stellar bars, much like our Milky Way, from eight to 11 billion years ago when the universe was much younger. This is the first time stellar bars have been observed in such young galaxies. Scientists identified two "barred" galaxies from about 11 billion years ago.



# API n°2 : Carbon intensity factor in the south of the UK

In [106]:
res_2 = requests.get('https://api.carbonintensity.org.uk/regional/intensity/2018-05-15T12:00Z/2018-05-16T12:00Z/postcode/RG10')
res_2 

<Response [200]>

In [121]:
uk = res_2.json()
uk
# co2

{'data': {'regionid': 12,
  'shortname': 'South England',
  'postcode': 'RG10',
  'data': [{'from': '2018-05-15T11:30Z',
    'to': '2018-05-15T12:00Z',
    'intensity': {'forecast': 235, 'index': 'moderate'},
    'generationmix': [{'fuel': 'biomass', 'perc': 0.2},
     {'fuel': 'coal', 'perc': 0},
     {'fuel': 'imports', 'perc': 0},
     {'fuel': 'gas', 'perc': 59.9},
     {'fuel': 'nuclear', 'perc': 0.4},
     {'fuel': 'other', 'perc': 0},
     {'fuel': 'hydro', 'perc': 0},
     {'fuel': 'solar', 'perc': 39.3},
     {'fuel': 'wind', 'perc': 0.2}]},
   {'from': '2018-05-15T12:00Z',
    'to': '2018-05-15T12:30Z',
    'intensity': {'forecast': 233, 'index': 'moderate'},
    'generationmix': [{'fuel': 'biomass', 'perc': 0.4},
     {'fuel': 'coal', 'perc': 0.1},
     {'fuel': 'imports', 'perc': 0},
     {'fuel': 'gas', 'perc': 58.9},
     {'fuel': 'nuclear', 'perc': 0.6},
     {'fuel': 'other', 'perc': 0},
     {'fuel': 'hydro', 'perc': 0},
     {'fuel': 'solar', 'perc': 39.8},
     {'fue

In [108]:
uk_df = pd.DataFrame(uk)
uk_df

data
data       [{'from': '2018-05-15T11:30Z', 'to': '2018-05-...
postcode                                                RG10
regionid                                                  12
shortname                                      South England

In [143]:
uk_df_norm = json_normalize(uk_df['data'][0])
# display(uk_df_norm)
print(uk_df_norm.max())

uk_df_norm['generationmix'][20]

from                  2018-05-16T11:30Z
to                    2018-05-16T12:00Z
intensity.forecast                  313
intensity.index                moderate
dtype: object


C:\Users\Emmanuel\AppData\Local\Temp\ipykernel_23412\2986634574.py:3: FutureWarning: Dropping of nuisance columns in DataFrame reductions (with 'numeric_only=None') is deprecated; in a future version this will raise TypeError.  Select only valid columns before calling the reduction.
  print(uk_df_norm.max())


[{'fuel': 'biomass', 'perc': 4},
 {'fuel': 'coal', 'perc': 1},
 {'fuel': 'imports', 'perc': 4.8},
 {'fuel': 'gas', 'perc': 73.8},
 {'fuel': 'nuclear', 'perc': 6},
 {'fuel': 'other', 'perc': 0},
 {'fuel': 'hydro', 'perc': 1.9},
 {'fuel': 'solar', 'perc': 0},
 {'fuel': 'wind', 'perc': 8.5}]

In [142]:
uk_df_norm_gene = json_normalize(uk_df_norm['generationmix'])
# uk_df_norm_gene

A department in the region of south of UK in consuming 73.8 % in gas and fuel as a source of energy

In [140]:
print(uk_df_norm['generationmix'][20])


[{'fuel': 'biomass', 'perc': 4}, {'fuel': 'coal', 'perc': 1}, {'fuel': 'imports', 'perc': 4.8}, {'fuel': 'gas', 'perc': 73.8}, {'fuel': 'nuclear', 'perc': 6}, {'fuel': 'other', 'perc': 0}, {'fuel': 'hydro', 'perc': 1.9}, {'fuel': 'solar', 'perc': 0}, {'fuel': 'wind', 'perc': 8.5}]


# API n°3 : NYC open data

In [148]:
res_3 = requests.get('https://data.cityofnewyork.us/resource/erm2-nwe9.json')

print(res_3)
ny = pd.DataFrame(res_3.json())

#data.shape

<Response [200]>


In [147]:
ny.columns

Index(['unique_key', 'created_date', 'agency', 'agency_name', 'complaint_type',
       'descriptor', 'location_type', 'incident_zip', 'incident_address',
       'street_name', 'cross_street_1', 'cross_street_2', 'address_type',
       'city', 'facility_type', 'status', 'resolution_description',
       'resolution_action_updated_date', 'community_board', 'bbl', 'borough',
       'x_coordinate_state_plane', 'y_coordinate_state_plane',
       'open_data_channel_type', 'park_facility_name', 'park_borough',
       'latitude', 'longitude', 'location', ':@computed_region_efsh_h5xi',
       ':@computed_region_f5dn_yrer', ':@computed_region_yeji_bk3q',
       ':@computed_region_92fq_4b7q', ':@computed_region_sbqj_enih',
       'intersection_street_1', 'intersection_street_2', 'landmark',
       'closed_date', 'bridge_highway_name', 'bridge_highway_direction',
       'road_ramp', 'bridge_highway_segment', 'taxi_pick_up_location',
       'taxi_company_borough'],
      dtype='object')

In [152]:
ny['bridge_highway_name'].value_counts().sum()

4

In [153]:
ny['bridge_highway_direction'].value_counts().sum()

3

we can see that in Ny we have 4 highways bridge and only 3 directions ..... odd !

In [154]:
ny['street_name'].value_counts().sum()

985

In [155]:
ny['incident_address'].value_counts().sum()

985

in this fact, we have as many street names than we have incidents !